src/components/realtime/index.ts

```diff
@@ -33,7 +33,14 @@ export class Realtime extends BaseComponent {
       return;
     }
 
-    const channel = new Channel(name, this.ioc, this.localParticipant);
+    let channel: Channel = this.channels.find((channel) => channel['name'] === name);
+
+    if (channel) {
+      this.logger.log(`Channel ${name} is already connected`);
+      return channel;
+    }
+
+    channel = new Channel(name, this.ioc, this.localParticipant);
```

Comment:
Use and map instead of array, this will avoid possible duplication, it will be easier to get a channel and more will have more performance than array